pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,287 @@
+package upgrade
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/weaveworks/pctl/pkg/bootstrap"
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/install"
+	"github.com/weaveworks/pctl/pkg/runner"
+	go_git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	gitssh "gopkg.in/src-d/go-git.v4/plumbing/transport/ssh"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type UpgradeValues struct {
+	RepoOrgAndName string
+	Remote         string
+	HeadBranch     string
+	BaseBranch     string
+	CommitMessage  string
+	Name           string
+	Namespace      string
+	ProfileBranch  string
+	ConfigMap      string
+	Out            string
+	ProfileRepoURL string
+	ProfilePath    string
+	GitRepository  string
+	Version        string
+}
+
+func Upgrade(upgradeValues UpgradeValues, w io.Writer) error {
+	config := config.GetConfigOrDie()
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Checking if entitlement exists...\n")
+
+	entitlement, err := getEntitlement(clientset)
+	if err != nil {
+		return err
+	}
+
+	upgradeValues.Name = "weave-gitops-enterprise"
+	upgradeValues.ProfileBranch = "main"
+	upgradeValues.ConfigMap = ""
+	upgradeValues.Out = "."
+	upgradeValues.ProfileRepoURL = "git@github.com:weaveworks/weave-gitops-enterprise-profiles.git"
+	upgradeValues.ProfilePath = "weave-gitops-enterprise"
+
+	repoURL, err := getRepoURL(upgradeValues.Remote)
+	if err != nil {
+		return err
+	}
+
+	if upgradeValues.RepoOrgAndName == "" {
+		githubRepoPath, err := getRepoOrgAndName(repoURL)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintf(w, "Deriving org/repo for PR as %v\n", githubRepoPath)
+		upgradeValues.RepoOrgAndName = githubRepoPath
+	}
+
+	if upgradeValues.GitRepository == "" {
+		gitRepositoryNameNamespace := "wego-system/" + strings.TrimSuffix(filepath.Base(repoURL), ".git")
```

Comment:
you'll need to use `upgradeValues.Namespace` instead of "wego-system" since user can pick another one.

Reply:
Yes right! 🙂 Thank you Luiz 🙏